tests/MemoryPack.Tests/Models/MultipleReferences.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace MemoryPack.Tests.Models;
+
+[MemoryPackable(GenerateType.MultipleReferences)]
+public partial class RefNode
+{
+    public RefNode() : this(Array.Empty<RefNode>())
+    {
+    }
+
+    [MemoryPackConstructor]
+    private RefNode(IReadOnlyList<RefNode> children)
+    {
+        Children = children;
+    }
+
+    public IReadOnlyList<RefNode> Children { get; private set; }
+
+    public void AddChild(RefNode child)
+    {
+        var list = Children?.ToList() ?? new List<RefNode>();
```

Comment:
This func allocates a bunch.

Calls toList() and then ToArray()?

Could you have an internal List<RefNode> that Children continues to be a IReadOnlyList<>